runtime/doc/msgpack_rpc.txt

```diff
@@ -270,10 +270,15 @@ a dictionary with these (optional) keys:
 				Defaults to false.
 
 Nvim will then send msgpack-rpc notifications, with the method name "redraw"
-and a single argument, an array of screen updates (described below).
-These should be processed in order. Preferably the user should only be able to
-see the screen state after all updates are processed (not any intermediate
-state after processing only a part of the array).
+and a single argument, an array of screen updates (described below).  These
+should be processed in order. Preferably the user should only be able to see
+the screen state after all updates in the same "redraw" event are processed
+(not any intermediate state after processing only a part of the array).
+
+New update kinds will be added in later versions of Nvim.  It is safe for a
+client to ignore new updates it doesn't know about.  Also, new parameters
+might be added to existing updates.  It is safe to ignore these as well, and
+just use the parameters known when the client was implemented.
```

Comment:
I can make the language stronger: clients _should_ ignore updates it doesn't know about.

> The general purpose of consolidating things in a single place is to train users to reach for that single place. If things are sprinkled here and there then people are less likely to also read the consolidated item.

The intention is certainly not to "sprinkle here and there". Right now nvim only has one attachable stream of events, so the information is relevant precisely here. When nvim gets multiple streams, we should establish shared conventions (like builtin events beginning with `nvim_`), and there will be section "event streams" or something in `msgpack_rpc.txt` (before specific streams like ui, buffers etc), and the shared concerns, including this paragraph, will be moved there.

> At the very least, why not link to |api-contract| ?

Of course, it just didn't exist the last time I pushed this PR.